doc/source_debugging.md

```diff
@@ -0,0 +1,75 @@
+# WAMR source debugging
+
+WAMR supports source level debugging based on DWARF, source map is not supported.
+
+## Build wasm application with debug information
+To debug your application, you need to compile them with debug information. You can use `-g` option when compiling the source code if you are using wasi-sdk (also work for emcc and rustc):
+``` bash
+/opt/wasi-sdk/bin/clang -g test.c -o test.wasm
+```
+
+Then you will get `test.wasm` which is a WebAssembly module with embedded DWARF sections. Further, you can use `llvm-dwarfdump` to check if the generated wasm file contains DWARF information:
+``` bash
+llvm-dwarfdump-12 test.wasm
+```
+
+## Debugging with interpreter
+
+1. Build iwasm with source debugging feature
```

Comment:
Add a step of how to build lldb with patch?

Reply:
Done